Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1

```diff
@@ -171,13 +171,62 @@ param (
         Write-Verbose "Unable to get the path for net.exe. Net command will be 'net'"
         return 'net'
     }
+
+    function Create-DestinationDirectory(
+        [string]$path
+    )
+    {
+        $psCredentialObject = New-Object PSCredential($credential.UserName, (ConvertTo-SecureString $credential.Password -AsPlainText -Force))
+        $destPath = $path
+        $foundParentPath = $false
+        $isRoot = $false
+
+        Write-Verbose "Creating path to directory: $path"
+        while($destPath -and (-not $foundParentPath))
+        {
+            try
+            {
+                New-PSDrive -Name WFCPSDrive -PSProvider FileSystem -Root $destPath -Credential $psCredentialObject
+                $foundParentPath = $true
+                Write-Verbose "Found parent path"
+                $relativePath = $path.Substring($destPath.Length)
+                New-Item -ItemType Directory WFCPSDrive:$relativePath -Force
+                Write-Verbose "Created directory"
+                Remove-PSDrive -Name WFCPSDrive
+            }
+            catch 
+            {
+                Write-Verbose "Caught exception: $_.Exception.Message"
+                $parentPath = Split-Path -Path $destPath -Parent
+                if(($parentPath.Length -eq 0) -and ($isRoot -eq $false))
+                {
+                    $destPath = [IO.Path]::DirectorySeparatorChar + [IO.Path]::DirectorySeparatorChar + ([System.Uri]($destPath)).Host
```

Comment:
Can you confirm that you don't need "\" in the end?

Reply:
"\\" does not seem to be required.